### PR TITLE
Add bottom bar hit error meter

### LINKED ISSUE #5932: Centered hit errors (like osu!stable)
I can't see the hit errors when it's set to the left or right side. It's further away from the sides than the bottom (Widescreen problem,it's not in the field of view) It would be nice if osu!lazer has the same hit error option as stable, as well as the scaling option.

One Option missed same on the stable:
ss (2019-08-30 at 08 16 00)

**Proposal designs of the feature:** 
Like this (is a screenshot montage, edited by paint.net)
ss (2019-08-30 at 08 29 02)

### PR DESCRIPTION
**Pull request by Game4all**
Closes #5932
Second attempt  using the approach proposed in #6025 

image

### CODE REVIEW

**Review by frenzibyte — CHANGES_REQUESTED** [review]
Can't this be calculated by a private function that spits the correct vector / anchor / axes? so you would give it the default positioned value of the type and the alignment and it would invert it or something, what do you think?

`getAligned(alignment, new Vector2(x, y))`:
```csharp
private Vector2 getAligned(Anchor alignment, Vector2 value) => alignment == Anchor.x1 ? value.Yx : value;
```

File: osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs [hunk 1]
```diff
@@ -233,29 +282,54 @@ public override void OnNewJudgement(JudgementResult judgement)
...
+                    RelativeSizeAxes = Axes.Y;
+                    RelativePositionAxes = Axes.X;
+                    Width = 3;
+                }
+                else
+                {
+                    RelativeSizeAxes = Axes.X;
+                    RelativePositionAxes = Axes.Y;
+                    Height = 3;
+                }
```
Comment (iiSaLMaN) [inline osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs]: These can be fairly simplified with:
```csharp
RelativeSizeAxes = isVertical ? Axes.Y : Axes.X;
RelativePositionAxes = isVertical ? Axes.X : Axes.Y;
Size = isVertical ? new Vector2(3, Size.Y) : new Vector2(Size.X, 3);
```

(making it more easier with what I proposal):
```csharp
RelativeSizeAxes = getAligned(alignment, Axes.X);
RelativePositionAxes = getAligned(alignment, Axes.Y);
Size = getAligned(alignment, new Vector2(Size.X, 3));
```

Comment (Game4all) [reply osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs]: I'm unsure about the utility of the code snippet you commented earlier as it is simply wrapping an inline conditionnal in a function. As for the properties , looks like I forgot to inline them

Comment (iiSaLMaN) [reply osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs]: > I'm unsure about the utility of the code snippet you commented earlier as it is simply wrapping an inline conditionnal in a function.

They make the code more readable tbh. I would implement this only if it's possible to and useful for the source code. (forgot to mention that)

File: osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs [hunk 2]
```diff
@@ -55,89 +56,112 @@ private void load(OsuColour colours)
...
+            else
+            {
+                InternalChild.Height = bar_height;
+                judgementsContainer.Width = judgement_line_width;
+                colourBars.Width = bar_width;
+                colourBarsEarly.Height = 0.5f;
+                colourBarsLate.Height = 0.5f;
+                arrowContainer.Width = chevron_size;
+                arrow.Y = 0.5f;
+            }
```
Comment (iiSaLMaN) [inline osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs]: There are also other checks like here for example. just to make it clear, `Width` and `Height` are components of the vector `Size`, `X` and `Y` are components of `Position`, so you can apply the same thing as I said above.

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs (hunk 2 above)
Comment (Game4all) [reply osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs]: Didn't know these were shorthand properties for accessing Size & Position properties

**Comment by Game4all** [comment]
Closing this as I don't intend to work further on this one.

**Pull request closed without merging**